server/controllers/user.js

```diff
@@ -0,0 +1,238 @@
+import groupUser from '../services/groupuser';
+import authenticate from '../helpers/authenticate';
+
+dotenv.config();
+const JWT_SECRET = process.env.JWT_SECRET;
+const salt = bcrypt.genSaltSync(10);
+/**
+ * class User: controls all user routes
+ * @class
+ */
+export default class User {
+  /**
+ * @description: controls a user's registration through route POST: api/v1/user/signup
+ * @param {Object} req request object
+ * @param {Object} res response object
+ * @return {Object} response containing the registered user
+ */
+  static signUp(req, res) {
+    const [username, password, email, phoneNumber] = [req.body.username,
+      req.body.password, req.body.email, req.body.phoneNumber];
+    const hashedPassword = bcrypt.hashSync(password, salt);
+    if (/^[a-zA-Z]{5,12}$/.test(username) === false) {
+      res.status(400).json({ message: 'Username should contain only letters and must have between 5-12 characters' })
+    } else {
+      user.saveUser(username, hashedPassword, email, phoneNumber, (users) => {
+        if (users instanceof Object) {
+          if (Array.isArray(users)) {
+            if (users[1] === false) {
+              res.status(409).json({ message: 'You already have an existing account. Kindly go and login' });
+            } else {
+              res.status(201).json({ message: 'Registration successful', user: { id: users[0].id, username: users[0].username, email: users[0].email } });
+            }
+          } else {
+            res.status(500).json({ message: 'Sorry, an unexpected error occurred' });
+          }
+        } else {
+          res.status(400).json({ message: users });
+        }
+      });
+    }
+  }
+
+  /**
+ * @description: controls a user's login through route POST: api/user/signin
+ * @param {Object} req request object
+ * @param {Object} res response object
+ * @return {Object} response containing the logged-in user
+ */
+  static signIn(req, res) {
+    const [username, password] = [req.body.username, req.body.password];
+    user.getUser(username, (users) => {
+      if (users.length === 0) {
+        res.status(404).json({ message: 'Invalid user!' });
+      } else if (bcrypt.compareSync(password, users[0].password)) {
+        const token = authenticate.generateToken({ username, id: users[0].id });
+        res.status(200).json({ message: 'You are now logged in', user: { id: users[0].id, user: username, email: users[0].email, token } });
+      } else {
+        res.status(401).json({ message: 'Incorrect password' });
+      }
+      if (users instanceof Object && !Array.isArray(users)) {
+        res.status(500).json({ message: 'Sorry, unexpected error occurred' });
+      }
+    });
+  }
+
+  /**
+ * @description: retrieves all users through route GET: api/users
+ * @param {Object} req request object
+ * @param {Object} res response object
+ * @return {Object} response containing all users
+ */
+  static getAllUsers(req, res) {
+    const userrs = req.headers.userrs;

```

Comment:
Was there a mistake in this variable name. I would suggest changing this to something straightforward and explanatory